R/summary_funcs.R

```diff
@@ -0,0 +1,78 @@
+        # GLM and Confound Settings
+          cat("\n--------\n Settings: \n")
+          cat(" -  Scheduler Name: ", gpa$scheduler, "\n")
+          cat(" -  TR: ", gpa$tr, "\n")
+          cat(" -  Multi-Level Run: ", gpa$multi_run[1], "\n")
+          cat(" -  GLM Settings: ", gpa$glm_settings, "\n")
+          if (!is.null(gpa$confound_settings$na_strings)) {
+            cat(" -  NA Strings: ", paste(gpa$confound_settings$na_strings, collapse = ", "), "\n")
+          }
+          if (!is.null(gpa$confound_settings$exclude_run)) {
+            cat(" -  Exclude Runs: ", gpa$confound_settings$exclude_run, "\n")
+          }
+          if (!is.null(gpa$confound_settings$exclude_subject)) {
+            cat(" -  Exclude Subjects: ", gpa$confound_settings$exclude_subject, "\n")
+          } 
+          if (!is.null(gpa$confound_settings$truncate_run)) {
+            cat(" -  Truncate Run: ", gpa$confound_settings$truncate_run, "\n")
+          }
+          if (!is.null(gpa$confound_settings$spike_volumes)) {
+            cat(" -  Spike Volumes: ", gpa$confound_settings$spike_volumes, "\n")
+          }
+          # VM/Key Columns
+          cat("--------\n Key Columns: \n")
+          cat("  ", paste(gpa$vm, collapse = ", "), "\n")
+          cat("n Expected Runs: ", gpa$n_expected_runs, "\n")
+          # Models: L1/2/3
+          cat("--------\n Models: \n")
+          if (!is.null(gpa$l1_models)) {
+            cat(" -  L1 Models: ", "Present \n")
+          } else {
+            cat(" -  L1 Models: NA\n")
+          }
+          if (!is.null(gpa$l1_models)) {
+            cat(" -  L2 Models: ", "Present \n")
+          } else {
+            cat(" -  L2 Models: NA\n")
+          }
+          if (!is.null(gpa$l1_models)) {
+            cat(" -  L3 Models: ", "Present \n")
+          } else {
+            cat(" -  L3 Models: NA\n")
+          }
+          # Computation Info
+          cat("--------\n Computation Info: \n")
+          cat(" FSL Level 1:\n")
+          cat("  -  Feat Time: ", "\n")
+          cat("  -  Feat Memory (GB): ", "\n")
+          cat(" FSL Level 2:\n")
+          cat("  -  Feat Time: ", "\n")
+          cat("  -  Feat Memory (GB): ", "\n")
+          cat(" FSL Level 3:\n")
+          cat("  -  Feat Time: ", "\n")
+          cat("  -  Feat Memory (GB): ", "\n")
+          # Output Directory and System Info
+          cat("--------\n System Info: \n")
+          cat(" -  Output Directory: ", gpa$output_directory[1], "\n")
+          cat(" -  System Name: ", gpa$sys_info[1], "\n")
+          cat(" -  Release: ", gpa$sys_info[2], "\n")
+          cat(" -  Version: ", gpa$sys_info[3], "\n")
+          cat(" -  Node Name: ", gpa$sys_info[4], "\n")
+          cat(" -  Machine: ", gpa$sys_info[5], "\n")
+          cat(" -  Login: ", gpa$sys_info[6], "\n")
+          cat(" -  User: ", gpa$sys_info[7], "\n")
+          cat(" -  Effective User: ", gpa$sys_info[8], "\n")
```

Comment:
I would also suggest adding info about the pipeline settings for software, specifically the `$compute_environment` field. This controls the lines added to relevant scripts to setup the environment at the time of compute:

```
$compute_environment
$compute_environment$global
[1] "module use /proj/mnhallqlab/sw/modules"

$compute_environment$r
[1] "module unload r"     "module load r/4.2.1"

$compute_environment$fsl
[1] "module unload fsl"     "module load fsl/6.0.4"

$compute_environment$afni
[1] "module unload afni"       "module load afni/23.0.07"
```